packages/paymaster/Readme.md

```diff
@@ -0,0 +1,183 @@
+# **Paymaster**
+
+ERC4337, Account abstraction, introduces the concept of Paymasters. These specialised entities play a pivotal role in revolutionising the traditional gas payment system in EVM transactions. Paymasters, acting as third-party intermediaries, possess the capability to sponsor gas fees for an account, provided specific predefined conditions are satisfied.
+
+## Installation
+
+Using `npm` package manager
+
+```bash
+npm i @biconomy/paymaster
+```
+
+OR
+
+Using `yarn` package manager
+
+```bash
+yarn add @biconomy/paymaster
+```
+
+**Usage**
+
+```typescript
+// This is how you create paymaster instance in your dapp's
+import { IPaymaster, BiconomyPaymaster } from '@biconomy/paymaster'
+
+  const paymaster = new BiconomyPaymaster({
+    paymasterUrl: '' // you can get this value from biconomy dashboard. https://dashboard.biconomy.io
+  })
+```
+
+**paymasterUrl** you can get this value from biconomy dashboard.
+
+Following are the methods that can be called on paymaster instance 
+
+```typescript
+export interface IHybridPaymaster<T> extends IPaymaster {
+  getPaymasterAndData(
+    userOp: Partial<UserOperation>,
+    paymasterServiceData?: T
+  ): Promise<PaymasterAndDataResponse>
+  buildTokenApprovalTransaction(
+    tokenPaymasterRequest: BiconomyTokenPaymasterRequest,
+    provider: Provider
+  ): Promise<Transaction>
+  getPaymasterFeeQuotesOrData(
+    userOp: Partial<UserOperation>,
+    paymasterServiceData: FeeQuotesOrDataDto
+  ): Promise<FeeQuotesOrDataResponse>
+}
+```
+
+**getPaymasterAndData**
+
+This function accepts a **`Partial<UserOperation>`** object that includes all properties of **`userOp`** except for the **`signature`** field. It returns **`paymasterAndData`** as part of the **`PaymasterAndDataResponse`**. The **`paymasterAndData`** string is signed by the paymaster's verifier signer, which will eventually result in the payment of your transaction fee by the paymaster.
+
+**buildTokenApprovalTransaction**
+
+This function is specifically used for token paymaster sponsorship. The primary purpose of this function is to create an approve transaction that gets batched with the rest of your transactions. This way, you will be paying the paymaster in ERC20 tokens, which will result in the paymaster paying on your behalf in native tokens.
```

Comment:
*token approval transaction
explain here spender will be paymaster

Reply:
"which will result in the paymaster paying on your behalf in native tokens" this is not needed and is rather confusing